src/cwe_checker_lib/src/checkers/cwe_476/context.rs

```diff
@@ -213,21 +214,25 @@ impl<'a> TaintAnalysis<'a> for Context<'a> {
     /// We assume that returning a tainted value means that the function may
     /// return a NULL pointer. This always generates a warning, even if this may
     /// be expected by the caller.
-    fn update_return(
+    fn update_return_callee(
         &self,
         state: &TaState,
         _call_term: &Term<Jmp>,
         return_term: &Term<Jmp>,
         calling_convention: &Option<String>,
-    ) {
-        if state.check_return_values_for_taint(
+    ) -> Option<TaState> {
+        if state.check_return_values_for_taint::<true>(
             self.vsa_result(),
             &return_term.tid,
             self.project,
             calling_convention,
         ) {
             self.generate_cwe_warning(&return_term.tid);
         }
+
+        // Keep analysis intraprocedural but do not force propagation to be
+        // stopped in the caller.
+        Some(TaState::new_empty())
```

Comment:
Might need to be adjusted if we adjust the other `update_return` functions (see above).